tiled/adapters/utils.py

```diff
@@ -1,15 +1,83 @@
+import math
 import warnings
-from typing import Any
+from typing import Any, Optional, Tuple
+
+import numpy as np
 
 # for back-compat
 from ..utils import node_repr as tree_repr  # noqa: F401
+from .type_alliases import NDSlice
 
 _MESSAGE = (
     "Instead of {name}_indexer[...] use {name}()[...]. "
     "The {name}_indexer accessor is deprecated."
 )
 
 
+def sliced_shape(shp: Tuple[int, ...], slc: Optional[NDSlice] = ...) -> Tuple[int, ...]:
```

Comment:
```py
In [1]: from ndindex import ndindex

In [2]: sliced_shape = ndindex(slice(0, 5)).newshape((10, 10))
```

Reply:
Note: we aleady have an `ndindex` dependency as we use it for exactly this situation, so it is a good bet.

Reply:
Fixed!